### PR TITLE
Unittesting/setup

### PR DESCRIPTION
**Pull request by nbreum15**

### CODE REVIEW

File: src/electric-mouse/Services/SectionService.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        }
+
+        /// <summary>
+        /// Adds a new section to the database.
+        /// </summary>
+        /// <param name="name">Name of the section.</param>
+        /// <param name="routeHallId">The id of the routeHall the section is in.</param>
+        public void AddSection(string name, int? routeHallId)
+        {
+            RouteSection section = new RouteSection { Name = name, RouteHall = _dbContext.RouteHalls.First(h => h.RouteHallID == routeHallId) };
```
Comment (jakobbsm) [inline src/electric-mouse/Services/SectionService.cs:28]: Line is too long, you should also split up your query. To make it even cleaner making a separate variable for routehall might be viable.

File: src/electric-mouse/Services/SectionService.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                section.Archived = true;
+                _dbContext.RouteSections.Update(section);
+                _dbContext.SaveChanges();
+            }
+        }
+
+        /// <summary>
+        /// Gets all the route sections in the database. It includes the routeHalls, routeSectionRelations and the routes connected to these relations.
+        /// </summary>
+        public List<RouteSection> GetAllRouteSections()
```
Comment (jakobbsm) [inline src/electric-mouse/Services/SectionService.cs:55]: Not sure about the readability on this one. Might be easier if it had a normal wrapper.

File: src/electric-mouse/Services/SectionService.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                          .ThenInclude(s => s.Route)
+                          .ToList();
+
+        /// <summary>
+        /// Archives all routes in the section by the given sectionId.
+        /// </summary>
+        /// <param name="sectionId">Id that matches the section where all the routes should be archived.</param>
+        public void ArchiveAllRoutesInSection(int? sectionId)
+        {
+            List<RouteSectionRelation> relations = _dbContext.RouteSectionRelations.Include(rs => rs.Route)
```
Comment (jakobbsm) [inline src/electric-mouse/Services/SectionService.cs:68]: New line

File: test/FileExtensionsTest.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using System.Threading.Tasks;
+using Xunit;
+using electric_mouse.Services;
+
+namespace test
+{
+    public class FileExtensionsTest
+    {
+        [Fact]
+        public void HasExtension_CheckForPNGInFileNameWithPNG_ReturnsTrue()
```
Comment (jakobbsm) [inline test/FileExtensionsTest.cs:13]: Shouldn't the class name make the suffix unnecessary?

File: test/project.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  "buildOptions": {
+    "debugType": "portable"
+  },
+  "dependencies": {
+    "xunit": "2.2.0-beta4-build3444",
+    "dotnet-test-xunit": "2.2.0-preview2-build1029",
+    "electric-mouse": {
+      "target": "project"
+    },
+    "moq.netcore": "4.4.0-beta8",
```
Comment (jakobbsm) [inline test/project.json]: Replace with: `"Moq": "4.6.38-alpha",` instead of .netcore version. Otherwise I don't seem to have any luck.

**Review by jakobbsm — CHANGES_REQUESTED** [review]

**Comment by nbreum15** [comment]
fuck u, not gonna happen bitch.

**Comment by jakobbsm** [comment]
Not fixing will lead to closes.

**Comment by jakobbsm** [comment]
We need it for master branch, so I'll let the changes pass for now. Look into them later though.

**Pull request merged**